### PR TITLE
Extract several libs to improve reusability, remove "core" dependency

### PR DESCRIPTION
**Pull request by proksch**
I made it easier to reuse and build-upon the FASTEN results by refactoring two major things in the code base:

- I have isolated several concerns, made them dedicated libraries, and moved them out of the repository.
- I have removed the dependency to FASTEN `core` project, which has introduced many, many transitive dependencies.

On top of these changes, I have also performed major house cleaning tasks, like updating dependencies or adapting build scripts.

### CODE REVIEW

File: infrastructure/infrastructure-impl/src/main/java/eu/f4sten/infra/impl/InfraConfig.java [hunk 1]
```diff
@@ -100,6 +100,25 @@ public PostgresConnector bindPostgresConnector() {
...
 
+    @Provides
+    @Singleton
+    public KafkaConnector bindKafkaConnector(RunnerArgs runnerArgs) {
+        assertFor(args) //
+                .notNull(a -> a.kafkaUrl, "kafka url") //
+                .that(a -> a.instanceId == null || !a.instanceId.isEmpty(), "instance id must be null or non-empty") //
+                .that(a -> a.kafkaGroupId == null || !a.kafkaGroupId.isEmpty(), "group id must be null or non-empty");
+
+        var pluginId = runnerArgs.run.replace("eu.f4sten.", "");
```
Comment (mir-am) [inline infrastructure/infrastructure-impl/src/main/java/eu/f4sten/infra/impl/InfraConfig.java:111]: Why is `eu.f4sten.` replaced with an empty string?

File: infrastructure/infrastructure/pom.xml [hunk 2]
```diff
@@ -11,37 +10,54 @@
 
     <dependencies>
         <dependency>
-            <groupId>com.google.inject</groupId>
-            <artifactId>guice</artifactId>
-            <version>5.0.1</version>
+            <groupId>dev.c0ps.franz</groupId>
```
Comment (mir-am) [inline infrastructure/infrastructure/pom.xml:13]: What is the `dev.c0ps.franz` pacakge? :)

File: plugins/pom-analyzer/pom.xml [hunk 3]
```diff
@@ -10,15 +9,64 @@
...
+        </dependency>
+
+        <!-- metadata db -->
+        <dependency>
+            <groupId>eu.fasten</groupId>
+            <artifactId>core-metadata</artifactId>
+            <version>0.0.19</version>
+        </dependency>
+        <dependency>
+            <!-- TODO remove this dependency (functionality should be native since jOOQ v3.15 -->
```
Comment (mir-am) [inline plugins/pom-analyzer/pom.xml:66]: Could you remove this unnecessary dependency in this PR as mentioned in the comment?

File: plugins/pom-analyzer/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/MavenResolvedArtifactImpl.java [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+    /**
+     * Maps an artifact to a file. This allows ShrinkWrap Maven resolver to package reactor related dependencies.
+     */
+    private static File artifactToFile(final Artifact artifact) throws IllegalArgumentException {
+        if (artifact == null) {
+            throw new IllegalArgumentException("ArtifactResult must not be null");
+        }
+
+        // FIXME: this is not a safe assumption, file can have a different name
```
Comment (mir-am) [inline plugins/pom-analyzer/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/MavenResolvedArtifactImpl.java:170]: Should this comment be addressed in another PR or in this one?

File: plugins/pom-analyzer/src/test/resources/PomExtractorTest/scm.pom
Comment (mir-am) [inline plugins/pom-analyzer/src/test/resources/PomExtractorTest/scm.pom:1]: Why are many of the `POMExtractorTest`s removed?

File: plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java [hunk 5]
```diff
@@ -69,7 +68,7 @@ void findCPayloadTest() throws IOException, JSONException {
         var message = new JSONObject(Files.readString(getTestResource("PayloadParsingTest/c_metadatadb_extension_message.json").toPath()));
         var payload = pp.findSourcePayload(message);
         assertNotNull(payload);
-        assertEquals(Constants.debianForge, payload.getForge());
+        assertEquals("debian", payload.getForge());
```
Comment (mir-am) [inline plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java:71]: I'd prefer to have the `Constants` class instead of the hard-coded strings.

**Review by mir-am — COMMENTED** [review]
Overall, the PR looks good to me.
I have some minor comments and questions.

File: infrastructure/infrastructure-impl/src/main/java/eu/f4sten/infra/impl/InfraConfig.java (hunk 1 above)
Comment (proksch) [reply infrastructure/infrastructure-impl/src/main/java/eu/f4sten/infra/impl/InfraConfig.java:111]: This is a simple means to make the logging easier to read. Instead of the full "eu.fasten.mavencrawler.Main", I cut off the `eu.fasten` part (And afaik also the `Main`). This is nto new functionality though, this has been there before.

File: infrastructure/infrastructure/pom.xml (hunk 2 above)
Comment (proksch) [reply infrastructure/infrastructure/pom.xml:13]: I have move the Kafka logic to a separate library. I called it "Franz" (homage to the writer Franz Kafka).

File: plugins/pom-analyzer/pom.xml (hunk 3 above)
Comment (proksch) [reply plugins/pom-analyzer/pom.xml:66]: I tried it, but it did not work without it... I am not an expert on jooq, so I left it in place for now.

File: plugins/pom-analyzer/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/MavenResolvedArtifactImpl.java (hunk 4 above)
Comment (proksch) [reply plugins/pom-analyzer/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/MavenResolvedArtifactImpl.java:170]: This file is literally a copy of a file from Shrinkwrap. I just added several smaller patches to get access to the resolved repository. I would prefer not to fix this in our own code base. If you think this is worthwhile looking into, then you should submit a patch to Shrinkwrap :P

File: plugins/pom-analyzer/src/test/resources/PomExtractorTest/scm.pom
Comment (proksch) [reply plugins/pom-analyzer/src/test/resources/PomExtractorTest/scm.pom:1]: The PomExtractor has been moved into the `dev.c0ps.commons-maven` packages, incl. its tests.

File: plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java (hunk 5 above)
Comment (proksch) [reply plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java:71]: I decided to get rid of the Constants class for three reasons:

- Dependency-wise, there is no elegant way right now that would allow sharing the Constants class between `data-processing` and `core`.
- I plan to phase out the other languages, as they are not maintained right now.
- I plan to refactor the `SourcesProvider` soon(tm) anyways, as I do not like that it extracts the jar contents.

I agree that magic strings are a *smell*, however, I see them as a temporary issue and they will vanish with further evolution of the code base anyways... Do you think I should still introduce a dedicated `FastenConstants` clone in the `data-processing` repo?

Comment (mir-am) [reply plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java:71]: I'd prefer that we introduce `FastenConstants` in the `data-processing` repo.
Also, I like to have the `SourcesProvider` plug-in in the repo. the JAR contents, i.e., Java source files, can be useful for training code language models.

Comment (proksch) [reply plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java:71]: Yes and no. I would like to have the sources as well, but from a resource usage perspective, it does not make any sense to unpack the .jar files. I will store the source jars and users that need the contents can unpack them and reduce them to the bits they need.

I will create a `FastenConstants` class.

Comment (mir-am) [reply plugins/sources-provider/src/test/java/eu/f4sten/sourcesprovider/utils/PayloadParsingTest.java:71]: Okay, it makes sense from the disk usage perspective.
I resolve the conversation.

File: plugins/maven-crawler/src/main/java/eu/f4sten/mavencrawler/MavenCrawlerArgs.java [hunk 6]
```diff
@@ -27,7 +27,6 @@ public class MavenCrawlerArgs {
     @Parameter(names = "--mc.kafkaOut")
     public String kafkaOut = DefaultTopics.INGEST;
 
-    @Parameter(names = "--mc.indexUrl")
-    public String indexUrl = "https://repo1.maven.org/maven2/.index/nexus-maven-repository-index.%d.gz";
-
+    @Parameter(names = "--mc.easyIndexUrl")
+    public String easyIndexUrl;
```
Comment (mir-am) [inline plugins/maven-crawler/src/main/java/eu/f4sten/mavencrawler/MavenCrawlerArgs.java:31]: Is it possible to the keep Maven URL as the default and also allow the user to override it if needed?

**Review by mir-am — APPROVED** [review]
Looks good to me.

File: plugins/maven-crawler/src/main/java/eu/f4sten/mavencrawler/MavenCrawlerArgs.java (hunk 6 above)
Comment (proksch) [reply plugins/maven-crawler/src/main/java/eu/f4sten/mavencrawler/MavenCrawlerArgs.java:31]: You are mixing up two different things! The "Easy Index" is an abstraction that I have created. It exposes the contents of the Maven index files via a convenient REST API. Reading the index requires a couple of outdated dependencies that I wanted to avoid in the data-processing repository, as they have affected the "freshness" of dependencies across other plugins as well. The URL that you are pointing out is the REST API of "Easy Index", no longer the path pattern to the index files.

**Review by mir-am — APPROVED** [review]
